CONTRIBUTING.md

```diff
@@ -0,0 +1,536 @@
+[`snake_case`](https://en.wikipedia.org/wiki/Snake_case).  Preprocessor
+macros and symbols should use `MACRO_CASE`:
+```c
+#define DOPPLER_EFFECT "Doppler Effect"
+
+static bool is_doppler_effect(char const *str);
+{
+        return strcmp(str, DOPPLER_EFFECT) == 0;
+}
+```
+
+##### Indentation
+Use an eight space indentation per code block level _without_ tabs, e.g.:
+```c
+#define MPTCP_GET_NL_ATTR(data, len, attr)                  \
+        do {                                                \
+                if (validate_attr_len(len, sizeof(*attr)))  \
+                        attr = data;                        \
+        } while(0)
+```
+```c
+void foo(void)
+{
+        // 8 spaces per indentation level, no tabs.
+        for (int i = 0; i < 20; ++i) {
+                printf("%d: ", i);
+
+                if (i % 2 == 0)
+                        printf("even\n");
+                else
+                        printf("odd\n");
+        }
+}
+```
+
+##### Line length
+Whenever possible, lines should not be wider than 74 columns:
+```c
+void fnord(struct mptcpd_nm const *network_monitor,
+           mptcpd_nm_callback callback,
+           void *data)
+{
+        // Line length no longer than 74 columns whenever possible.
+        for (int i = 0; i < 20; ++i) {
+                if (i % 2 == 0) {
+                        // String literal split to fit within 74 columns.
+                        printf("%d: The quick brown fox jumps over "
+                               "the lazy dog\n",
+                               i);
+                } else {
+                        /*
+                          Place all function arguments on their own
+                          line if line wrap is necessary for them.
+                        */
+                        mptcpd_nm_foreach_interface(network_monitor,
+                                                    callback,
+                                                    data);
+                }
+        }
+}
+```
+
+##### Parameter Alignment
+Function parameters should be aligned on the column immediately after the
+opening paranethesis, or split one per line if they cannot all fit on
+a 74 column line.  If after splitting across multiple lines any of the
+parameters extends beyone column 74 shift all parameters down one line
+to the next indentation level:
+```c
+bool hyperbolic_parabaloid_equal(struct hyperbolic_paraboloid const *lhs,
+                                 struct hyperbolic_paraboloid const *rhs);
+
+struct hyperbolic_paraboloid const *get_hyperbolic_parabaloid(
+       struct topology const *t,
+       int *error);
+```
+The same alignment applies to arguments in function calls, control
+statements, etc.
+
+##### Brace Placement
+* Structure declarations and function definitions
+Place the opening brace on the line below the structure or function
+name.  Align the opening and closing brace with the first character in
+the structure or function name.
```

Comment:
This doesn't get formatted quite right, looks like "Structure declarations and function definitions Place the opening brace" (no line break, all same style).

Viewed formatted file at https://github.com/ossama-othman/mptcpd/blob/contributing/CONTRIBUTING.md#brace-placement

Reply:
Good catch.  The GitHub Flavored Markdown viewer browser plugin I used for local viewing formats the text as expected.  I'll adjust the text to work with GitHub accordingly.